sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RandomIdGenerator.java

```diff
@@ -27,12 +27,11 @@ public String generateSpanId() {
   @Override
   public String generateTraceId() {
     long idHi;
-    long idLo;
     ThreadLocalRandom random = ThreadLocalRandom.current();
     do {
       idHi = random.nextLong();
-      idLo = random.nextLong();
-    } while (idHi == INVALID_ID && idLo == INVALID_ID);
+    } while (idHi == INVALID_ID);
+    long idLo = random.nextLong();
     return TraceId.fromLongs(idHi, idLo);
```

Comment:
Can be compacted even more now by TraceId.fromLongs(idHi, random.nextLong()), although this negatively impacts readability (same for android).

Reply:
Hehe - yeah I'm mostly interested in "real code" (unnecessary check) vs "lines of code" (local variable name that provides some good semantic context too).